testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ProducerStubFactory.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.testkit
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.function.{ Function => JFunction }
+
+import akka.Done
+import akka.actor.{ ActorRef, ActorSystem, Props }
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Flow, Source }
+import com.lightbend.lagom.internal.testkit.{ InternalSubscriberStub, TopicBufferActor }
+import com.lightbend.lagom.scaladsl.api.broker.Topic.TopicId
+import com.lightbend.lagom.scaladsl.api.broker.{ Subscriber, Topic }
+
+import scala.concurrent.Future
+
+/**
+ * Factors [[com.lightbend.lagom.scaladsl.testkit.ProducerStub]]'s.
+ */
+final class ProducerStubFactory(actorSystem: ActorSystem, materializer: Materializer) {
+
+  private val topics = new ConcurrentHashMap[String, ProducerStub[_]]
+
+  def producer[T](clzz: Class[T], topicId: String): ProducerStub[T] = {
```

Comment:
Don't pass `Class` arguments in a Scala API - use `ClassTag` type classes instead. In this case, `clzz` isn't even used, so you don't even need a `ClassTag`, just remove the class parameter altogether. Users should just:

```scala
val stub = producerStubFactory.producer[MyMessage]("my-topic")
```